uk_bin_collection/uk_bin_collection/councils/MidSuffolkDistrictCouncil.py

```
@@ -51,34 +41,31 @@ def parse_data(self, page: str, **kwargs) -> dict:
-            # `__clone` / OOM crash inside the Grid container. Now we exit as
-            # soon as options are available.
+            # Wait for the address dropdown to appear and populate.
             def _populated_select(d):
-                selects = d.find_elements(By.CSS_SELECTOR, "select")
-                for s in selects:
-                    if len(s.find_elements(By.TAG_NAME, "option")) > 1:
-                        return s
-                return False
+                try:
+                    select_el = d.find_element(By.ID, f"{_NS}uprn")
+                except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)